### PR TITLE
Mock WordPress.com: Move to flags

### PR DESCRIPTION
**Pull request by alshakero**
### Changes

Follow up to https://vertexp2.wordpress.com/2024/06/17/mock-wordpress-com-in-30-seconds/#comment-2059

### Testing steps
1. Run `yarn`.
2. Run `yarn start-mock-wordpress-com`.
3. Calypso should serve at `https://wordpress.com:443`.

### Backwards compat
1. Run `bin/mock-wordpressdotcom`.
2. It should tell `Please use yarn start-mock-wordpress-com`

### CODE REVIEW

File: bin/mock-wordpressdotcom.sh [hunk 1]
```diff
@@ -1,36 +1,4 @@
...
-
-    yarn > /dev/null
-
-    echo "Done!"
-
-    echo "Make sure to:"
-    echo "  1. Add ${GREEN}127.0.0.1 wordpress.com${NC} to your hosts file."
-    echo "  2. Restart Calypso with ${GREEN}yarn start${NC}."
-    echo "  3. Type ${GREEN}thisisunsafe${NC} in Chrome when you visit wordpress.com."
-    echo "  4. Unmock when done by calling this script with ${GREEN}--unmock${NC}"
```
Comment (tyxla) [inline bin/mock-wordpressdotcom.sh:24]: I believe this was useful and we'll need to add it somewhere for future reference. Folks might get blocked by the browser's invalid certificate warning for example.

File: client/server/boot/index.js [hunk 2]
```diff
@@ -31,6 +32,15 @@ export default function setup() {
...
 		require( 'calypso/server/bundler' )( app );
 
+		// When mocking WordPress.com to point locally, wordpress.com/wp-login.php will hit Calypso.
+		// redirect traffic to de.wordpress.com to hit the real backend.
+		if ( process.env.MOCK_WORDPRESSDOTCOM === '1' ) {
+			app.use(
+				'/wp-login.php',
+				createProxyMiddleware( { target: 'https://de.wordpress.com/wp-login.php' } )
+			);
+		}
```
Comment (tyxla) [inline client/server/boot/index.js:43]: How does this work? Won't `https://de.wordpress.com/wp-login.php` redirect to `https://wordpress.com/log-in/de` itself?

I think logging in just won't work in this context. Perhaps we should just add a mechanism to bypass some routes and mark them as "non-working in this mocked/sandboxed setup"?

File: client/server/boot/index.js [hunk 3]
```diff
@@ -31,6 +32,15 @@ export default function setup() {
 	if ( 'development' === process.env.NODE_ENV ) {
 		require( 'calypso/server/bundler' )( app );
 
+		// When mocking WordPress.com to point locally, wordpress.com/wp-login.php will hit Calypso.
+		// redirect traffic to de.wordpress.com to hit the real backend.
+		if ( process.env.MOCK_WORDPRESSDOTCOM === '1' ) {
+			app.use(
+				'/wp-login.php',
```
Comment (tyxla) [inline client/server/boot/index.js:40]: Should we also handle attempts to go to `/log-in`?

**Review by tyxla — COMMENTED** [review]
Works well overall 👍 Thanks for working on it, and for your patience.

I think we could work a bit on the documentation before landing this - part of what you're removing was useful and I'd keep it inline and in `/docs` perhaps - to ensure that folks understand that they need to:

* Sandbox the domain to their localhost
* Force the browser to accept the invalid SSL certificate

Left a few more questions, let me know what you think.

File: client/server/boot/index.js (hunk 2 above)
Comment (alshakero) [reply client/server/boot/index.js:43]: I think it only redirects GET requests! It works for POST requests. You can login using this method just fine.

Comment (tyxla) [reply client/server/boot/index.js:43]: Got it - that's a clever hack! It's worth adding a comment to explain what we're doing.

Comment (alshakero) [reply client/server/boot/index.js:43]: There is a comment. Would you like me to improve on it?

File: client/server/boot/index.js (hunk 3 above)
Comment (alshakero) [reply client/server/boot/index.js:40]: /log-in works without any issues.

File: client/server/boot/index.js (hunk 2 above)
Comment (tyxla) [reply client/server/boot/index.js:43]: I think that's clear enough actually - thanks!

**Review by tyxla — APPROVED** [review]
Looks great, go for it, thanks @alshakero 🚀

**Pull request merged**